### ISSUE TITLE
Feature Request: Mark individual as paid

### ISSUE THREAD

**Issue by MAPReiff** [eneiluj/cospend-nc#36]
I would like a way to settle individual bills (not just the entire project) as well as a way to settle individual members of the bill. Not everyone can/will pay at the same time, and currently, the only real option is to settle the bill and undo specific users who still owe money.

Could these features potentially be implemented?

**Comment by eneiluj** [comment]
> I would like a way to settle individual bills (not just the entire project)

What do you mean by "settle individual bills"? As bills have effect on the global balances, I feel there is no need to make actions about individual bills. Only the balance is considered when talking about settlement.
What's the use case there?

> as well as a way to settle individual members of the bill

Do you mean "members of the project"?
Bringing the balance of one member to zero can be done by simply creating a reimbursement bill. What would be the added value of a special settlement feature for one user?

It could be interesting to allow selecting a sub list of members involved in the settlement. The generated reimbursement plan would only bring those members to a zero balance. What do you think?

**Comment by MAPReiff** [comment]
> What do you mean by "settle individual bills"? As bills have effect on the global balances, I feel there is no need to make actions about individual bills. Only the balance is considered when talking about settlement.
> What's the use case there?

If a project has multiple bills in it, everyone may not pay them all off at once. The ability to settle individual bills and not all bills in the project at once would solve this issue.

> Do you mean "members of the project"?
> Bringing the balance of one member to zero can be done by simply creating a reimbursement bill. What would be the added value of a special settlement feature for one user?

I meant members of the bill itself, as each bill doesn't have to have all project members selected. The added value of this is to track when individual people pay their own share of a bill. The current setup requires the entire thing to be settled, and then undo the member show haven't paid yet, which isn't a very elegant solution. The added value I guess is an easier way to make an individual as paid.

> It could be interesting to allow selecting a sub list of members involved in the settlement. The generated reimbursement plan would only bring those members to a zero balance. What do you think?

That would work very well for settling projects (I 100% think this should be added), but I would also like to see settlements per bill, not just per project if possible (also with the above sub list settlement idea).

**Comment by eneiluj** [comment]
I think we don't see shared budget management the same way :grin:. The goal of Cospend is to avoid thinking of the bills. Bills are created, then the balances are computed, then you see who's positive and negative.
The exact bill history does not matter, the reimbursement is completely abstracted from the bills and from the payers thanks to the balances.

Settling a bill does not make sense to me. That's the sum of all bills (represented by the balance) that makes sense.

"Settling a bill" means deleting one bill or creating an opposite one. It does not seem to require a "feature". You can do it manually. Partially settling a bill seems weird. Why would you do that?

If you manage expenses and reimbursement on a "per bill" basis, you expect people to reimburse each bill independently. Cospend can avoid that and even avoid reimbursement at all if all balances are staying close to zero because the expenses are nicely shared between member over time.

> If a project has multiple bills in it, everyone may not pay them all off at once.

A Cospend bill is supposed to be something that has been payed by one person for a subset of the others. If this person hasn't payed/bought the thing already, then the bill does not exist.
The goal is to represent the debts between each member and the whole group. There is no obligation to reimburse right after creating a bill. A negative balance means one owes the group.

If you want to make partial settling, it is still possible to manually create reimbursement bills that represent a money transation between 2 members and changes their balance. For example if someone has a huge negative balance and wants to give money to someone to bring his/her balance closer to zero.

**Comment by MAPReiff** [comment]
> Settling a bill does not make sense to me. That's the sum of all bills (represented by the balance) that makes sense.

This would imply that people are willing to (or have enough money to) pay off all of their debts at one time. While in an ideal world this would always be possible, some people may not be in a situation where they can settle off of their debts at once.

> "Settling a bill" means deleting one bill or creating an opposite one. It does not seem to require a "feature". You can do it manually. Partially settling a bill seems weird. Why would you do that?

Settling a bill would be for if the one who is owed money for that bill receives all payments for that one specific bill. It would be easier to settle the one bill rather than the entire project and then undo specific people if everyone cannot pay off all of their bills at once.

> If you want to make partial settling, it is still possible to manually create reimbursement bills that represent a money transation between 2 members and changes their balance. For example if someone has a huge negative balance and wants to give money to someone to bring his/her balance closer to zero.

This is what I would like to do, but instead of manually reimbursing people, simply having an easy button to mark someone as paid and not paid would simplify the entire process.

Example Scenario:
Four people (Person A, Person B, Person C and Person D) are assigned to a project which has multiple bills.
Person A has $250 in their bank account
Person B has $200 in their bank account
Person C has $50 in their bank account
Person D has $75 in their bank account

Person A makes a $200 purchase, which is split evenly between all four people. People B, C and D now owe A $50 each.
Person B makes a $100 purchase, which is split evenly between all four people. People A, C and D now owe B $25 each.
Person C now owes a total of $75, however, he only has $50 in his account. Because of this, he cannot pay both Person A a and Person B at the same time. He decides to send Person A the $50 he owes and will pay Person B the $25 at a later date when he has more money.

With the current system, the solution would be to settle the entire project and then undo the settlement for Person C on that one bill he still owes, which is not a very elegant solution. What I am asking for is a way to simply mark individual people as paid (or settle them individually on either a bill or project), as manually undoing the settlement for a specific person on a specific bill will undoubtedly lead to confusion and clutter in the bill history log.

**Comment by CoreyBurger** [comment]
I think this is a UI issue rather than a request. Part of the issue is as a new user, I found to no "record payment" option. I would reword "Project settlement" to "Record payment" and also move it out of the three dot menu. Ideally it should be a button on each project. And then reuse the same UI to show who paid whom, with the amounts below.

I think that would answer the original request to record partial payment

**Comment by la-balisa** [comment]
> Person C now owes a total of $75, however, he only has $50 in his account. Because of this, he cannot pay both Person A a and Person B at the same time. He decides to send Person A the $50 he owes and will pay Person B the $25 at a later date when he has more money.
> 
> With the current system, the solution would be to settle the entire project and then undo the settlement for Person C on that one bill he still owes, which is not a very elegant solution. What I am asking for is a way to simply mark individual people as paid (or settle them individually on either a bill or project), as manually undoing the settlement for a specific person on a specific bill will undoubtedly lead to confusion and clutter in the bill history log.

Hi ! With the current system, a good solution is to create a new "bill" for this reimbursement. You define Person C as the payer, Person A as the only member concerned by the bill and 50$ as the amount. Then Person C will be shown with a -25$ balance, and Person B still have a +25$ balance.
You can even categorize this as "reimbursement" (i think, i use it in french) so it won't count in the total spent money for this project.


> I meant members of the bill itself, as each bill doesn't have to have all project members selected. The added value of this is to track when individual people pay their own share of a bill.

I think this is a really is paradigm issue. With cospend, you never reimburse your share of a specific bill ; the calculations are done project-wide. We don't care if you reimburse 50$ for bill 1 or bill 2. You owe 75$, you already paid 50$ and still have to pay 25$ to the right people, that is all that matters.
